src/presentation_exchange.rs

```diff
@@ -447,9 +466,8 @@ pub struct ConstraintsField {
     purpose: Option<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
     name: Option<String>,
-    // TODO: JSONSchema validation at deserialization time
     #[serde(skip_serializing_if = "Option::is_none")]
-    filter: Option<SchemaValidator>,
+    filter: Option<serde_json::Value>,
```

Comment:
Replacing `SchemaValidator` to use a generic `Value` type and using `jsonschema` to perform the validation.

SchemaValidator provides helpful methods to construct a filter object and will perform some validation, but it is not as feature rich as `jsonschema`.

Consider removing `SchemaValidator` from the code, unless there is a reason to keep it.

Reply:
I would be in favour of removing it, I think we eventually need a more featureful builder than what it is anyway.

Reply:
I'll wire up the schema validation in #8 and remove the schema validator from this PR.